### PR TITLE
Fix the site editor Admin Bar menu item

### PR DESCRIPTION
**Pull request by youknowriad**
Follow-up to #61741 

## What?

Now that the details page has been removed entirely, we need to direct the admin bar menu item to the "edit" mode page directly. 

## Testing Instructions

1- Open the frontend of your website (block theme)
2- Click that the "site editor" link
3- The active template should be opened in the site editor in edit mode.

### CODE REVIEW

**Review by Mamaduka — APPROVED** [review]
Thanks, Riad!

File: lib/compat/wordpress-6.6/admin-bar.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+					'postType' => 'wp_template',
+					'postId'   => $_wp_current_template_id,
+					'canvas'   => 'edit',
+				),
+				admin_url( 'site-editor.php' )
+			),
+		)
+	);
+}
+remove_action( 'admin_bar_menu', 'wp_admin_bar_edit_site_menu', 40 );
```
Comment (Mamaduka) [inline lib/compat/wordpress-6.6/admin-bar.php:47]: Prevents doing extra work in core.

File: backport-changelog/6.6/61266.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+https://github.com/WordPress/wordpress-develop/pull/6605
```
Comment (ellatrix) [inline backport-changelog/6.6/61266.md]: file name needs to be 6605

Comment (ellatrix) [reply backport-changelog/6.6/61266.md]: > Please update the content of backport-changelog/6.6/61266.md to include the core backport PR URL, or update the file name to match the core backport PR number.

Comment (youknowriad) [reply backport-changelog/6.6/61266.md]: Ok I though it was the Gutenberg PR number.

**Comment by Mamaduka** [comment]
Can we merge this?

**Comment by youknowriad** [comment]
Yeah forgot about it :P

**Pull request merged**